src/test/scala/vectorpipe/vectortile/WeightedCentroid.scala

```diff
@@ -0,0 +1,57 @@
+package vectorpipe.vectortile
+
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.expressions.MutableAggregationBuffer
+import org.apache.spark.sql.expressions.UserDefinedAggregateFunction
+import org.apache.spark.sql.jts.PointUDT
+import org.apache.spark.sql.types._
+
+class WeightedCentroid extends UserDefinedAggregateFunction {
+
+  // Defind the schema of the input data
```

Comment:
*Define

Reply:
That's what I get for pasting code off the internet.

Reply:
Maybe you should unfind it?